src/DistributedMetadata/CatalogService.cpp

```diff
@@ -100,9 +100,11 @@ void CatalogService::doBroadcast()
 
     /// Default max_block_size is 65505 (rows) which shall be bigger enough for a block to contain
     /// all tables on a single node
+
+    /// [NOTICE] the `tables` must be selected to ensure that select result set has at least one row for each node
```

Comment:
Revise the statement to 
```
/// We include "system.tables" and / or "system.tasks" tables in the resulting block on purpose . 
/// It is to avoid an empty block if there are no production tables in the system, which will cause 
/// consistency problem in CatalogService (like the last deleted table does not get deleted from CatalogService)
```